src/skeleplex/graph/skeleton_graph.py

```diff
@@ -126,6 +130,49 @@ def get_next_node_key(graph: nx.Graph) -> int:
     return int(np.max(graph.nodes)) + 1 if graph.nodes else 0
 
 
+def flip_spline(spline: B3Spline, path: np.ndarray) -> B3Spline:
+    """Recomputes the spline inverse to the path."""
+    return spline.from_points(path[::-1]), path[::-1]
+
+
+def orient_splines(graph: nx.DiGraph) -> nx.DiGraph:
+    """Checks if the splines are oriented correctly.
+
+    If the beginning of the spline is closer to the end node than the start node,
+    it gets flipped.
+    Also checks if the edge coordinates are aligend with the spline.
+    This only checks, if the splines are correctly connected to the nodes,
+    not the order in the Graph. Best used on a directed graph.
+
+    Parameters
+    ----------
+    graph : nx.DiGraph
+        The graph to orient the splines in
+
+    """
+    for u, v, attr in graph.edges(data=True):
+        spline = attr[EDGE_SPLINE_KEY]
+        u_coord = graph.nodes[u][NODE_COORDINATE_KEY]
+        spline_coordinates = spline.eval(np.array([0, 1]))
+        # check if spline evaluation is closer to the start or end node
+        if np.linalg.norm(u_coord - spline_coordinates[0]) > np.linalg.norm(
+            u_coord - spline_coordinates[-1]
+        ):
+            logger.info(f"Flipped spline of edge {u,v}.")
+            edge_coordinates = attr[EDGE_COORDINATES_KEY]
+            # check if path is inverse to spline
+            if np.linalg.norm(
+                edge_coordinates[0] - spline_coordinates[0]
+            ) < np.linalg.norm(edge_coordinates[-1] - spline_coordinates[-1]):
+                edge_coordinates = edge_coordinates[::-1]
+
+            flipped_spline, flipped_cords = flip_spline(spline, edge_coordinates)
+            graph[u][v][EDGE_SPLINE_KEY] = flipped_spline
+            graph[u][v][EDGE_COORDINATES_KEY] = flipped_cords
```

Comment:
Should this be under the `if np.linalg.norm(edge_coordinates[0] - spline_coordinates[0] ...` statement? It seems like you only want to flip the spline if that condition is met.

Also, a minor point: it might be a good idea to store all of the updated edge properties and set them at the end once you've looped through all edges. If you update in-place as you go through the loop and there is an error in the loop, you end up with an object that is partially processed.

It's also possible that it is faster. I think I read the updating the edge/node properties in batch is faster (e.g., with `nx.set_node_attributes`), but I'm not 100% sure.

Reply:
I got two if statements, one checks if the spline is flipped and the other one just checks if the path is parallel to the spline, in case something went wrong at some point. Usually this should always be the case, but I thought it wouldn't hurt